include/triton/core/tritonbackend.h

```diff
@@ -1359,6 +1360,86 @@ TRITONBACKEND_ISPEC TRITONSERVER_Error* TRITONBACKEND_ModelInstanceExecute(
     const uint32_t request_count);
 
 
+/// [BETA] Below is an experimental feature and the interface is subject to
+/// changes.
+///
+
+typedef enum TRITONBACKEND_attributetype_enum {
+  TRITONBACKEND_ATTRIBUTE_PREFERRED_INSTANCE_GROUP,
+} TRITONBACKEND_AttributeType;
+
+/// Query the backend for different model attributes. This function is optional,
+/// a backend is not required to implement it. This function is called when
+/// Triton requires further backend / model information to perform operations.
+/// This function may be called multiple times within the lifetime of the
+/// backend (between TRITONBACKEND_Initialize and TRITONBACKEND_Finalize), and
+/// optional 'model' may be provided if it is called for a particular model.
+/// The backend may return error to indicate that the specific attribute does
+/// not apply, otherwise, 'backend_attributes' should be created and set to
+/// reflect such attribute and 'nullptr' should be returned. Triton will take
+/// the ownership of 'backend_attributes' if 'nullptr' is returned.
+///
+/// \param backend The backend.
+/// \param model The model. 'nullptr' may be provided if no model is associated.
+/// \param type The attribute to query about.
+/// \param in_params The array of input parameters as a complement of the
+/// attribute inquiry.
+/// \param in_count The number of input parameters.
+/// \param backend_attributes The backend attribute. See
+/// TRITONBACKEND_BackendAttribute on how it should be set for different 'type'.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_ISPEC TRITONSERVER_Error* TRITONBACKEND_GetBackendAttribute(
```

Comment:
I was wondering do you think there will be any benefits if we add an API to set this callback instead of adding it like `TRITONBACKEND_Initialize/TRITONBACKEND_Finalize`?

Reply:
I think "adding an API to set this callback" will then require the backend to connect the dots, i.e. if the backend does provide backend attribute, the control will need to reach the backend before they can register the callback. In the current way, Triton can access the "callback" when loading the shared library.